reporter/base_reporter.go

```diff
@@ -160,27 +160,37 @@ func (b *baseReporter) FrameMetadata(args *FrameMetadataArgs) {
 		if sourceFile == "" {
 			// The new SourceFile may be empty, and we don't want to overwrite
 			// an existing filePath with it.
-			if s, exists := (*frameMap)[addressOrLine]; exists {
-				sourceFile = s.FilePath
+			if source, exists := (*frameMap).GetAndRefresh(addressOrLine,
+				pdata.FrameMapLifetime); exists {
+				sourceFile = source.FilePath
 			}
 		}
 
-		(*frameMap)[addressOrLine] = samples.SourceInfo{
+		(*frameMap).AddWithLifetime(addressOrLine, samples.SourceInfo{
```

Comment:
Isn't `AddWithLifetime` redundant here as we've previously set the lifetime with `SetLifetime` ?

```suggestion
		(*frameMap).Add(addressOrLine, samples.SourceInfo{
```

Reply:
That is true, see [here](https://github.com/elastic/go-freelru/blob/ef66858fb8b9aab9e78eab7feeac9e59b3194a72/lru.go#L439-L443)

The comment for the `Add()` function should clarify this. Created the issue https://github.com/elastic/go-freelru/issues/86 .